opencog/attention/SimpleImportanceDiffusionAgent.cc

```diff
@@ -100,10 +100,12 @@ void SimpleImportanceDiffusionAgent::updateMaxSpreadPercentage() {
         Handle h = resultSet.front();
         h->getIncomingSet(back_inserter(resultSet));
         h = resultSet.front();
-        resultSet = h->getOutgoingSet();
-        h = resultSet.back();
-        resultSet = h->getOutgoingSet();
-        h = resultSet.front();
+        if (h->isLink()) {
```

Comment:
Not sure why these `isLink()` tests are needed.  Presumably, its already known that the handle is a link, and thus, that asking for its outgoing set is a valid operation.  If there are cases where the handle is not a link, then .. well there's a bug in the code, and this test serves only to hide that bag, make it invisible, make it harder to find.

But whatever. A hack to patch things up for now is OK, I guess. I will merge this.